language/compiler/bytecode_source_map/src/disassembler.rs

```diff
@@ -0,0 +1,330 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::mapping::SourceMapping;
+use crate::source_map::SourceName;
+use failure::prelude::*;
+use libra_types::identifier::IdentStr;
+use vm::access::ModuleAccess;
+use vm::file_format::{
+    FieldDefinitionIndex, FunctionDefinition, FunctionDefinitionIndex, Kind, SignatureToken,
+    StructDefinition, StructDefinitionIndex, StructFieldInformation, TableIndex, TypeSignature,
+};
+
+/// Holds the various options that we support while disassembling code.
+#[derive(Debug, Default)]
+pub struct DisassemblerOptions {
+    /// Only print public functions
+    pub only_public: bool,
+
+    // TODO: Need to implement this.
+    /// Print the bytecode for the instructions within the function.
+    pub print_code: bool,
+}
+
+impl DisassemblerOptions {
```

Comment:
btw this entire impl seems superfluous to me given the struct is public and btw I like the public struct as well.
Just a note but do as you wish

Reply:
Good call! I'll delete